### PR TITLE
MEN-4671: mender-client: New package mender-modules-gen and Update Module tests 

### PR DESCRIPTION
**Pull request by lluiscampos**
This is an alternative approach for #1352 . See specifically discussion at https://github.com/mendersoftware/meta-mender/pull/1352#discussion_r625143882

### CODE REVIEW

**Comment by lluiscampos** [comment]
Testing together with https://github.com/mendersoftware/mender/pull/750 @ [Pipeline 296739574](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296739574)

File: meta-mender-core/recipes-mender/mender-client/mender-client.inc [hunk 1]
```diff
@@ -1,7 +1,14 @@
 DESCRIPTION = "Mender tool for doing OTA software updates."
 HOMEPAGE = "https://mender.io"
 
+BBCLASSEXTEND = "native"
+RDEPENDS_${PN}_class-native = ""
+DEPENDS_class-native = ""
+
+PACKAGES_append = " ${PN}-modules-gen"
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/mender-client.inc:8]: AFAIK, it's not required to use `${PN}` here, so you could call it mender-modules-gen. I think that name is a bit better, but no strong preference.

**Review by kacf — APPROVED** [review]

File: meta-mender-core/recipes-mender/mender-client/mender-client.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-client/mender-client.inc:8]: Thanks, changed. I also think is a better name.

**Comment by lluiscampos** [comment]
> Testing together with [mendersoftware/mender#750](https://github.com/mendersoftware/mender/pull/750) @ [Pipeline 296739574](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296739574)

New build @ [Pipeline 296812352](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296812352)

**Comment by lluiscampos** [comment]
Testing with https://github.com/mendersoftware/mender/pull/750 and https://github.com/mendersoftware/mender-qa/pull/500 @ [Pipeline 296906012](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296906012)

**Comment by lluiscampos** [comment]
> Testing with [mendersoftware/mender#750](https://github.com/mendersoftware/mender/pull/750) and [mendersoftware/mender-qa#500](https://github.com/mendersoftware/mender-qa/pull/500) @ [Pipeline 296906012](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296906012)

New build @ [Pipeline 296962621](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296962621)

**Comment by lluiscampos** [comment]
> > Testing with [mendersoftware/mender#750](https://github.com/mendersoftware/mender/pull/750) and [mendersoftware/mender-qa#500](https://github.com/mendersoftware/mender-qa/pull/500) @ [Pipeline 296906012](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296906012)
> 
> New build @ [Pipeline 296962621](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/296962621)

**Last** build @ [Pipeline 297173533](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/297173533)

**Pull request merged**